ner/nopd_pib_reports.py

```diff
@@ -0,0 +1,85 @@
+import json
+import random
+import spacy
+from spacy.util import minibatch, compounding
+from spacy.training import Example
+import pandas as pd
+import deba
+import pandas as pd
+
+
+def read_pdfs():
+    pdfs = pd.read_csv(deba.data("ocr/nopd_pib_reports_pdfs.csv"))
+    return pdfs
+
+
+def spacy_model(pdfs):
+    ###  import labeled_data that has been exported from doccano
+
+    # labeled_data = []
+    # with open(
+    #     r"data/raw/new_orleans_pd/training_data/nopd_pib_reports.jsonl",
+    #     "r",
+    # ) as read_file:
+    #     for line in read_file:
+    #         data = json.loads(line)
+    #         labeled_data.append(data)
+
+    # TRAINING_DATA = []
+    # for entry in labeled_data:
+    #     entities = []
+    #     for e in entry["label"]:
+    #         entities.append((e[0], e[1], e[2]))
+    #     spacy_entry = (entry["data"], {"entities": entities})
+    #     TRAINING_DATA.append(spacy_entry)
+
+    ### train model
+
+    nlp = spacy.load(deba.data("raw/new_orleans_pd/model/nopd_pib_reports.model"))
+    # nlp = spacy.blank("en")
+    # ner = nlp.create_pipe("ner")
+    # nlp.add_pipe("ner")
+    # ner.add_label("allegation")
+
+    # optimizer = nlp.begin_training()
+    # for itn in range(100):
+    #     random.shuffle(TRAINING_DATA)
+    #     losses = {}
+    #     for batch in spacy.util.minibatch(
+    #         TRAINING_DATA, size=compounding(3.0, 1.5, 1.001)
+    #     ):
+    #         for text, annotations in batch:
+    #             doc = nlp.make_doc(text)
+    #             example = Example.from_dict(doc, annotations)
+    #             nlp.update([example], sgd=optimizer, losses=losses, drop=0.4)
+    #             print(losses)
+
+    # save model to disk:
+    # nlp.to_disk(deba.data("raw/new_orleans_pd/model/nopd_pib_reports.model"))
```

Comment:
I see a lot of similarities in the commented out NER code compared to [ner/post_officer_history_reports.py](https://github.com/ppact/processing/blob/2cd172505cd308107d4dd8db4e83219d31f067cd/ner/post_officer_history_reports.py). Better extract them out to a library function.

And the training should be turned on/off via a make variable (e.g. `make TRAIN=y`) rather than commenting out code. What do you think?

Reply:
Extracted the what was previously the `spacy_model` function into its sub-parts: `train_spacy_model` and `load_spacy_model`. Is commenting out the `train_spacy_model` function sufficient for this purpose? Please review.